.github/workflows/pr.yml

```diff
@@ -16,6 +16,27 @@ jobs:
       - run: yarn --frozen-lockfile
       - run: yarn ${{ matrix.step }}
 
+  test-playwright:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v3
+      - run: yarn --frozen-lockfile
+
+      # Cache Playwright browsers
+      - name: Cache Playwright browsers
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/ms-playwright
+          key: playwright-browsers-${{ runner.os }}-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            playwright-browsers-${{ runner.os }}-
+
+      # Install Playwright browsers if not cached
+      - run: yarn playwright install
+      - run: yarn build
+      - run: yarn test:playwright
```

Comment:
nit: this whole file can just call the `build_and_test` workflow vs. having to duplicate. see an example of that here - https://github.com/ar-io/ar-io-sdk/blob/main/.github/workflows/release.yml#L14-L16